src/interface/extension_interface.hpp

```diff
@@ -46,6 +50,61 @@ struct get_second_step_op<MeanOperator> {
   using type = AddOperator;
 };
 
+/**
+ * @brief Implementation of matrix copy operators for transpose cases with out
+ * of place memory copy.
+ */
+template <bool in_place, bool trans, typename sb_handle_t, typename element_t,
+          typename index_t, typename in_t, typename out_t>
+typename std::enable_if<trans && !in_place, typename sb_handle_t::event_t>::type
```

Comment:
Did you consider unifying the `_matcopy_impl` and using `if` (this is normally done in `src/interface/blas2_interface.hpp`) rather than `enable_if`?